hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGetInfoHandler.java

```diff
@@ -58,13 +80,117 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var tokenStore = context.createStore(ReadableTokenStore.class);
+        final var op = query.tokenGetInfoOrThrow();
+        if (op.hasToken()) {
+            final var token = tokenStore.get(requireNonNull(op.token()));
```

Comment:
Instead of requireNonNull(op.token()) can do op.tokenOrThrow()

Reply:
Since if (op.hasToken()) already true, op.token() is not null. However, I added requireNonNull here to just get rid of Sonar warning.